tests/system/src/test/java/test/robot/javafx/stage/StageFocusTest.java

```diff
@@ -0,0 +1,141 @@
+
+    static final Color SCENE_COLOR = Color.LIGHTGREEN;
+
+    private Stage theStage = null;
+
+    @BeforeAll
+    public static void setupOnce() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        eventReceivedLatch = new CountDownLatch(1);
+    }
+
+    /**
+     * Checks whether Stage is actually shown when calling show()
+     *
+     * Meant as a "canary" test of sorts to ensure other tests relying on
+     * Stage being actually shown and on foreground work fine.
+     */
+    @Test
+    public void testStageHasFocusAfterShow() throws InterruptedException {
+        Util.runAndWait(() -> {
+            theStage = getStage(false);
+
+            Group root = new Group();
+            Scene scene = new Scene(root, STAGE_SIZE, STAGE_SIZE);
+            scene.setFill(SCENE_COLOR);
+            scene.setOnKeyPressed(e -> {
+                if (e.getCode() == KeyCode.A) {
+                    eventReceivedLatch.countDown();
+                }
+            });
+
+            theStage.setScene(scene);
+            theStage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> {
+                Platform.runLater(() -> {
+                    theStage.setX(STAGE_X);
+                    theStage.setY(STAGE_Y);
+                    startupLatch.countDown();
+                });
+            });
+            theStage.show();
+        });
+
+        assertTrue(startupLatch.await(TIMEOUT, TimeUnit.MILLISECONDS), "Timeout waiting for test stage to be shown");
+
+        // check if isFocused returns true
+        assertTrue(
+            theStage.isFocused(),
+            "Stage.isFocused() returned false! Stage does not have focus after showing. Some tests might fail because of this. " +
+            "If that is the case, try re-running the tests with '--no-daemon' flag in Gradle."
+        );
+
+        // give UI a bit of time to finish showing transition
+        // ex. on Windows above latch is set despite the UI still "animating" the show
+        sleep(500);
+
+        // check if window is on top
+        Util.runAndWait(() -> {
+            WritableImage capture = getRobot().getScreenCapture(null, STAGE_X, STAGE_Y, STAGE_SIZE, STAGE_SIZE, false);
+            PixelReader captureReader = capture.getPixelReader();
+            for (int x = 0; x < STAGE_SIZE; ++x) {
+                for (int y = 0; y < STAGE_SIZE; ++y) {
+                    Color color = captureReader.getColor(x, y);
+                    assertColorEquals(SCENE_COLOR, color, TOLERANCE);
+                }
+            }
+        });
+
+        // check if we actually have focus and key presses are registered by the app
+        Util.runAndWait(() -> {
+            getRobot().keyPress(KeyCode.A);
+        });
+        assertTrue(
+            eventReceivedLatch.await(TIMEOUT, TimeUnit.MILLISECONDS),
+            "Event received latch timed out! Stage most probably did not have focus after showing. Some tests might fail because of this. " +
```

Comment:
I'd suggest to replace "Event received latch timed out!", or re-phrase completely, maybe

"The Stage did not received the key stroke generated by Robot.  This might happen if the stage did not receive focus.  Some tests might fail because of this.
Try re-running the tests with '--no-daemon' flag in Gradle.".

or something along these lines.